src/TokenGenerator.php

```diff
@@ -0,0 +1,145 @@
+<?php declare(strict_types=1);
+
+namespace Pdsinterop\Solid\Auth;
+
+use Pdsinterop\Solid\Auth\Utils\Jwks;
+use Pdsinterop\Solid\Auth\Enum\OpenId\OpenIdConnectMetadata as OidcMeta;
+
+class TokenGenerator
+{
+    ////////////////////////////// CLASS PROPERTIES \\\\\\\\\\\\\\\\\\\\\\\\\\\\
+
+    /** @var Config */
+    public $config;
+
+    //////////////////////////////// PUBLIC API \\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\
+
+    final public function __construct(
+        Config $config
+    ) {
+        $this->config = $config;
+    }
+	
+	public function generateRegistrationAccessToken($clientId, $privateKey) {
+		$issuer = $this->config->getServer()->get(OidcMeta::ISSUER);
+
+		// Create JWT
+		$signer = new \Lcobucci\JWT\Signer\Rsa\Sha256();
+		$keychain = new \Lcobucci\JWT\Signer\Keychain();				
+		$builder = new \Lcobucci\JWT\Builder();
+		$token = $builder
+			->setIssuer($issuer)
+            ->permittedFor($clientId)
+			->set("sub", $clientId)
+			->sign($signer, $keychain->getPrivateKey($privateKey))
+			->getToken();
+		$result = $token->__toString();
```

Comment:
As the line below returns the result, this line can be removed.

Reply:
agreed, fixed.